helix-term/src/commands.rs

```diff
@@ -2191,227 +2190,215 @@ fn global_search(cx: &mut Context) {
         path: PathBuf,
         /// 0 indexed lines
         line_num: usize,
+        line_content: String,
     }
 
     impl FileResult {
-        fn new(path: &Path, line_num: usize) -> Self {
+        fn new(path: &Path, line_num: usize, line_content: String) -> Self {
             Self {
                 path: path.to_path_buf(),
                 line_num,
+                line_content,
             }
         }
     }
 
-    impl ui::menu::Item for FileResult {
-        type Data = Option<PathBuf>;
-
-        fn format(&self, current_path: &Self::Data) -> Row {
-            let relative_path = helix_stdx::path::get_relative_path(&self.path)
-                .to_string_lossy()
-                .into_owned();
-            if current_path
-                .as_ref()
-                .map(|p| p == &self.path)
-                .unwrap_or(false)
-            {
-                format!("{} (*)", relative_path).into()
-            } else {
-                relative_path.into()
-            }
-        }
+    struct GlobalSearchConfig {
+        smart_case: bool,
+        file_picker_config: helix_view::editor::FilePickerConfig,
     }
 
     let config = cx.editor.config();
-    let smart_case = config.search.smart_case;
-    let file_picker_config = config.file_picker.clone();
+    let config = GlobalSearchConfig {
+        smart_case: config.search.smart_case,
+        file_picker_config: config.file_picker.clone(),
+    };
 
-    let reg = cx.register.unwrap_or('/');
-    let completions = search_completions(cx, Some(reg));
-    ui::raw_regex_prompt(
-        cx,
-        "global-search:".into(),
-        Some(reg),
-        move |_editor: &Editor, input: &str| {
-            completions
-                .iter()
-                .filter(|comp| comp.starts_with(input))
-                .map(|comp| (0.., std::borrow::Cow::Owned(comp.clone())))
-                .collect()
-        },
-        move |cx, _, input, event| {
-            if event != PromptEvent::Validate {
-                return;
-            }
-            cx.editor.registers.last_search_register = reg;
+    let columns = vec![
+        PickerColumn::new("path", |item: &FileResult, _| {
+            helix_stdx::path::get_relative_path(&item.path)
+                .to_string_lossy()
+                .into_owned()
+                .into()
+        }),
+        PickerColumn::new("contents", |item: &FileResult, _| {
+            item.line_content.as_str().into()
+        })
+        .without_filtering(),
```

Comment:
I think this column is vey occasionally useful. If I have no idea where the match is that I'm looking for, for example a function definition in a big C codebase, having the line is a nice way to eliminate results visually quickly. It probably would need to be trimmed though so the whitespace doesn't take up the whole table.

This does look fairly bad most of the time though and we can't highlight the matched parts because it isn't fuzzy matched. I'll add a `PickerColumn::hidden` constructor for this